packages/osd-dev-utils/src/simple_opensearch_dashboards_platform_plugin_discovery.ts

```diff
@@ -49,10 +49,10 @@ export function simpleOpenSearchDashboardsPlatformPluginDiscovery(
         (acc: string[], dir) => [
           ...acc,
           resolve(dir, '*/opensearch_dashboards.json'),
-          resolve(dir, '*/*/opensearch_dashboards.json'),
-          resolve(dir, '*/*/*/opensearch_dashboards.json'),
-          resolve(dir, '*/*/*/*/opensearch_dashboards.json'),
-          resolve(dir, '*/*/*/*/*/opensearch_dashboards.json'),
+          resolve(dir, '*/!(build)/opensearch_dashboards.json'),
+          resolve(dir, '*/!(build)/*/opensearch_dashboards.json'),
+          resolve(dir, '*/!(build)/*/*/opensearch_dashboards.json'),
+          resolve(dir, '*/!(build)/*/*/*/opensearch_dashboards.json'),
```

Comment:
It kind of strikes me as weird that we only exclude `build` directories at the second of five levels. So `foo/bar/build/baz/` will match but not `foo/build/bar/baz`. Not an objection, it just feels a little fragile.

Reply:
`/plugins` expect to have `build` inside a folder named after the plugin and we want to exclude `/plugins/*/build` only. We place the `build` at that depth from `/plugins`